### ISSUE TITLE
Allow script to transform flag input via callback

### ISSUE THREAD

**Issue by jrmykolyn** [sindresorhus/meow#81]
**WHAT**
Allow dependent scripts to (optionally) specify how flag input should be parsed by providing `callback` functions at instantiation time.

Callback functions would be applied to the raw CLI input, and the resulting values would be exposed to the dependent script as normal (eg. via `cli.flags`).

For example:
```
meow( {
    flags: {
        unicorn: {
            alias: 'u',
            callback: value => {
                // Perform transformation and return result. 
            }
        }
    }
} );
```

**WHY**
This would be useful for cases where the underlying API requires an array value for a specific option.

Rather than updating the API to support/parse both arrays and strings, the transformation could be handled directly in the CLI layer.

For example:

```
meow( {
    flags: {
        excludes: {
            callback: value => {
                return value.split(',');
            }
        }
    }
} );
```

```
$ <cli command> <input> --excludes logs,temp,dist // [ 'logs', 'temp', 'dist' ]
```

**HOW**
I've stubbed out an [initial implementation](https://github.com/jrmykolyn/meow/commit/fa35fe337b63d520ad5dbbb68953f9216c65a7bf) and test on my fork, which I'd be happy to submit for review.

**NOTES**
I stumbled on Issue #44 *after* digging into this proposed update. I'm not sure what the state of that issue is, but it looks like there's at least *some* overlap in terms of functionality.

**Comment by sindresorhus** [comment]
Yeah, I think this could be useful. I would name it `parse` instead of `callback` though.

> Rather than updating the API to support/parse both arrays and strings,

I think it would be useful to support an `array` type. The problem is that there are two conventions for specifying arrays. Either `foo --array=1 --array=2` or `foo --array=1,2`.

**Comment by jrmykolyn** [comment]
@sindresorhus Good points!

```
The problem is that there are two conventions for specifying arrays
```

If both conventions are well known/understood, I think the second example (flag name appears once, followed by comma delimited values) wins out in terms of usability. Alternatively, would it be possible to support both styles?

*If* `meow` is updated to support arrays, will it be possible to enforce a specific type on the values?

**Comment by jrmykolyn** [comment]
Anything I can do to help?

**Comment by sindresorhus** [comment]
I am going to decline adding a per-flag parse callback in meow. It adds a lot of API and typing complexity for a very small wrapper, and it overlaps with behavior already available in full parsers. For now, I recommend doing the transformation after parsing, or using a higher level parser if you need custom coercion. If you want native comma-separated or multi-value support at the parser level, please open an issue in yargs-parser, since meow is a thin wrapper around it.
